tests/compiletest.rs

```diff
@@ -39,6 +39,25 @@ fn run_tests(mode: Mode, path: &str, target: Option<String>) -> Result<()> {
         flags.push(target.into());
     }
 
+    // If we're on linux, then build the shared object file for testing external C function calls.
+    if cfg!(target_os = "linux") {
+        let cc = option_env!("CC").unwrap_or("cc");
+        Command::new(cc)
+            .args([
+                "-shared",
+                "-o",
+                "tests/extern-so/libtestlib.so",
+                "tests/extern-so/test.c",
+                // Only add the functions specified in libcode.version to the shared object file.
+                // This is to avoid automatically adding `malloc`, etc.
+                // Source: https://anadoxin.org/blog/control-over-symbol-exports-in-gcc.html/
+                "-fPIC",
+                "-Wl,--version-script=tests/extern-so/libcode.version",
+            ])
+            .output()
```

Comment:
You need to check `status` for success, otherwise this just checks whether `cc` was executed, not whether it succeeded